# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: 0f43d0f3d542
# Full Hash: 0f43d0f3d542e22b92f4609576f050d77391a206
# Author: Kelsey Gilbert <jgilbert@mozilla.com>
# Date: 2021-12-15 21:51:13
# Description:
#   Bug 1745308 - Only warn once if ClientWaitSync cannot return availability. r=gfx-reviewers,nical
#   
#   This avoids spamming warnings across IPC that sometimes causes shmem
#   exhaustion.
#   
# ==============================================================================

diff -r 63b2fd522aa8 -r 0f43d0f3d542 dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Wed Dec 15 09:37:17 2021 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Wed Dec 15 09:39:57 2021 +0000
@@ -5119,9 +5119,12 @@
   const bool canBeAvailable =
       (sync.mCanBeAvailable || StaticPrefs::webgl_allow_immediate_queries());
   if (!canBeAvailable) {
-    EnqueueWarning(
-        "ClientWaitSync must return TIMEOUT_EXPIRED until control has"
-        " returned to the user agent's main loop.");
+    if (!sync.mHasWarnedNotAvailable) {
+      EnqueueWarning(
+          "ClientWaitSync must return TIMEOUT_EXPIRED until control has"
+          " returned to the user agent's main loop. (only warns once)");
+      sync.mHasWarnedNotAvailable = true;
+    }
     return LOCAL_GL_TIMEOUT_EXPIRED;
   }
 